runtime-modules/content/src/lib.rs

```diff
@@ -1435,227 +1453,320 @@ decl_module! {
 
             // Build a new thread
             let new_thread = Thread_ {
                 title_hash: title_hash,
-                category_id: category_id,
-                author_id: forum_user_id,
-                cleanup_pay_off: BalanceOf::<T>::zero(),
+                author_id: member_id,
+                bloat_bond: thread_init_bloat_bond,
                 number_of_posts: T::PostId::zero(),
-                channel_id: channel_id,
+                channel_id: params.channel_id,
             };
 
             // Store thread
-            <ThreadById<T>>::mutate(category_id, new_thread_id, |value| {
+            <ThreadById<T>>::mutate(new_thread_id, |value| {
                 *value = new_thread.clone()
             });
 
             // Add inital post to thread
             let _ = Self::add_new_post(
                 new_thread_id,
-                category_id,
                 text_hash,
-                forum_user_id,
+                member_id,
+                account_id,
+                first_post_init_bloat_bond,
+                params.post_mutable,
             );
 
             // Update next thread id
          <NextThreadId<T>>::mutate(|n| *n += One::one());
 
-         // increase category's thread counter
-         <CategoryById<T>>::mutate(category_id, |category| category.num_direct_threads += 1);
-
             // Generate event
             Self::deposit_event(
                 RawEvent::ThreadCreated(
                     new_thread_id,
-                    forum_user_id,
-                    category_id,
-                    title_hash,
-            channel_id,
+                    member_id,
+                    params,
                 )
             );
 
             Ok(())
      }
 
-    #[weight = 10_000_000]
-    fn delete_thread(
+       #[weight = 10_000_000]
+       fn delete_thread(
             origin,
-            forum_user_id: ForumUserId<T>,
-            category_id: T::CategoryId,
+            actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
             thread_id: T::ThreadId,
         ) -> DispatchResult {
             // Ensure data migration is done
-            // Self::ensure_data_migration_done()?;
+           // Self::ensure_data_migration_done()?;
+
+           let account_id = ensure_signed(origin)?;
+           let thread = Self::ensure_thread_exists(&thread_id)?;
+           let channel_id = thread.channel_id;
 
-            let account_id = ensure_signed(origin)?;
 
             Self::ensure_can_delete_thread(
                 &account_id,
-                &forum_user_id,
-                &category_id,
-                &thread_id
+                &actor,
+                &channel_id,
+                &thread_id,
             )?;
 
-            //
-            // == MUTATION SAFE ==
-            //
+           // iterate over values that share the first key
+            let iter = <PostById<T>>::iter_prefix_values(thread_id);
```

Comment:
This is not a safe way to delete a thread, please re-read `Cleanup` section of https://github.com/Joystream/joystream/issues/2545.